### PR TITLE
helm: Make auth type configurable

### PR DESCRIPTION
**Pull request by agurgel-te**
This allows a Helm user to configure the authentication type used by `teleport-cluster`. In my case, I need GitHub.

Similar to #7450.

### CODE REVIEW

File: examples/chart/teleport-cluster/values.yaml [hunk 1]
```diff
@@ -15,6 +15,10 @@ clusterName: ""
 # Version of teleport image, if different from chart version in Chart.yaml.
 teleportVersionOverride: ""
 
+# Default authentication type. Possible values are 'local' and 'github' for OSS and 'oidc', 'saml' and 'false' for Enterprise.
```
Comment (webvictim) [inline examples/chart/teleport-cluster/values.yaml:18]: `local` and `github` also work for Enterprise so this clarifies that a bit.

```suggestion
# Default authentication type. Possible values are 'local' and 'github' for OSS, plus 'oidc', 'saml' and 'false' for Enterprise.
```

**Review by webvictim — COMMENTED** [review]
@antonio-te Looks great - could you please also add `authenticationType` to the correct section of https://github.com/gravitational/teleport/blob/master/docs/pages/kubernetes-access/helm/reference.mdx ?

Once that's done I can approve.

**Comment by jonaskello** [comment]
I was looking for exactly this. The only other way I find to do it would be to run the chart in custom mode but it seems that is not a recommended mode. What is stopping the merge now?

**Comment by webvictim** [comment]
I’ve asked the author to add the value to the Helm chart reference in our docs. Once that’s done we can approve and merge it.

**Comment by webvictim** [comment]
@antonio-te Please take a look at my previous comment.

**Comment by antonio-te** [comment]
I could have sworn I force-pushed a new commit including the suggested change.

**Review by webvictim — APPROVED** [review]

**Comment by jonaskello** [comment]
Nicely done :-). I would like to use this as soon as possible. What frequency are the helm charts are published at? I cannot seem to find any helm chart newer than 6 which seems old since teleport 7.0.3 has ben release?

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**